### PR TITLE
Feature/error handling

### PR DESCRIPTION
**Pull request by cbmackintosh**
## What does this PR do?

- Intalls Cypress
- Adds error handling function to the APIcalls file
- Adds 400 and 500 error handling logic
- Introduces Switch components to the App component render method

## How should it be tested in the UI?

- Cypress was installed as a dev dependency. Not sure if you'll need to install it locally when you fetch this branch
- Use Cypress to stub network requests and see error message response
- You can also test bad paths by manually navigating to `localhost3000/anything` to test the Switch catch all.

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -21,19 +22,23 @@ class App extends React.Component<Props> {
...
+      savedNotes: [],
+      error: ''
     };
   }
 
   componentDidMount() {
-    getApis().then((data) => this.setState({ apiList: data.entries, currentApis: data.entries }));
+    getApis()
+    .then((data) => this.setState({ apiList: data.entries, currentApis: data.entries }))
+    .catch(err => this.setState({ error: err.message }))
```
Comment (Jacksonmcguire) [inline src/components/App/App.tsx:41]: Good call on the catch, had completely forgotten about that until now

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -126,50 +131,58 @@ class App extends React.Component<Props> {
...
+                      <section className='saved-notes'>
+                        <h3>Notes</h3>
+                        {savedNotes}
+                      </section>
+                    </main>
+                );
+              }
+            }}
+          />
+          <Route render={() => <ErrorMessage statusCode={"404"}/>} />
```
Comment (Jacksonmcguire) [inline src/components/App/App.tsx:181]: Sweet, definitely think a switch like this is the best for an error page like this

**Review by Jacksonmcguire — COMMENTED** [review]
```
undefined
TypeScript error in undefined(undefined,undefined):
Cannot find type definition file for 'cypress'.
  The file is in the program because:
    Entry point of type library 'cypress' specified in compilerOptions  TS2688
```
Could just be something on my end, but got this error when trying to open up the fetched branch. Code wise though everything looks good

File: cypress/integration/metAPI-testing.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      cy.contains('500')
+    })
+
+    it('test bad path', () => {
+      cy.visit('http://localhost:3000/nonexistent-page');
+      cy.contains('400')
+    })
+
+  })
+
```
Comment (itsnameissteven) [inline cypress/integration/metAPI-testing.ts:37]: looks good

File: src/components/App/App.tsx (hunk 1 above)
Comment (itsnameissteven) [reply src/components/App/App.tsx:41]: I like it, nice and clean

**Review by itsnameissteven — COMMENTED** [review]
This looks good. i'll merge so we can all start writing tests

**Pull request merged**